Source/WebCore/css/StyleProperties.cpp

```diff
@@ -576,12 +577,34 @@ String StyleProperties::fontValue() const
     if (auto shorthand = isSingleFontShorthand())
         return getValueNameAtomString(shorthand.value());
 
+    // Font stretch values can only be serialized as keywords.
+    // While it's not clear the CSS font specification calls for this, web platform tests expect us to convert percentages to keywords.
+    CSSValueID fontStretchPercentageAsKeyword = CSSValueInvalid;
+    if (int fontStretchPropertyIndex = findPropertyIndex(CSSPropertyFontStretch); fontStretchPropertyIndex != -1) {
+        if (auto value = dynamicDowncast<CSSPrimitiveValue>(*propertyAt(fontStretchPropertyIndex).value()); value && value->isPercentage()) {
+            auto valueAsDouble = value->doubleValue();
+            FontSelectionValue valueAsFontSelectionValue { static_cast<float>(valueAsDouble) };
+            auto valueAfterRoundTrip = valueAsFontSelectionValue;
+            if (valueAsDouble != valueAfterRoundTrip)
+                return emptyString();
```

Comment:
question: What transformation by FontSelectionValue can cause this to be hit?

Reply:
It would be nice to add a comment describing that, test would be nice, but optional.

Reply:
Added a comment and a test.